src/topology/bounded_continuous_function.lean

```diff
@@ -22,32 +22,51 @@ universes u v w
 variables {α : Type u} {β : Type v} {γ : Type w}
 
 /-- The type of bounded continuous functions from a topological space to a metric space -/
-def bounded_continuous_function (α : Type u) (β : Type v) [topological_space α] [metric_space β] :
+structure bounded_continuous_function
+  (α : Type u) (β : Type v) [topological_space α] [metric_space β] extends continuous_map α β :
   Type (max u v) :=
-{f : α → β // continuous f ∧ ∃C, ∀x y:α, dist (f x) (f y) ≤ C}
+(bounded' : ∃C, ∀x y:α, dist (to_fun x) (to_fun y) ≤ C)
 
-local infixr ` →ᵇ `:25 := bounded_continuous_function
+localized "infixr ` →ᵇ `:25 := bounded_continuous_function" in bounded_continuous_function
 
 namespace bounded_continuous_function
 section basics
 variables [topological_space α] [metric_space β] [metric_space γ]
 variables {f g : α →ᵇ β} {x : α} {C : ℝ}
 
-instance : has_coe_to_fun (α →ᵇ β) :=  ⟨_, subtype.val⟩
+instance : has_coe_to_fun (α →ᵇ β) :=  ⟨_, λ f, f.to_fun⟩
+
+protected lemma bounded (f : α →ᵇ β) : ∃C, ∀x y:α, dist (f x) (f y) ≤ C := f.bounded'
+protected lemma continuous (f : α →ᵇ β) : continuous f := f.to_continuous_map.continuous
 
 lemma bounded_range : bounded (range f) :=
-bounded_range_iff.2 f.2.2
+bounded_range_iff.2 f.bounded
 
-/-- If a function is continuous on a compact space, it is automatically bounded,
-and therefore gives rise to an element of the type of bounded continuous functions -/
-def mk_of_compact [compact_space α] (f : α → β) (hf : continuous f) : α →ᵇ β :=
-⟨f, hf, bounded_range_iff.1 $ bounded_of_compact $ compact_range hf⟩
+/-- A continuous function with an explicit bound is a bounded continuous function. -/
+def mk_of_bound (f : C(α, β)) (C : ℝ) (h : ∀ x y:α, dist (f x) (f y) ≤ C) : α →ᵇ β :=
+⟨f, ⟨C, h⟩⟩
+
+@[simp] lemma mk_of_bound_coe {f} {C} {h} : (mk_of_bound f C h : α → β) = (f : α → β) :=
+rfl
+
+/-- A continuous function on a compact space is automatically a bounded continuous function. -/
+def mk_of_compact [compact_space α] (f : C(α, β)) : α →ᵇ β :=
+⟨f, bounded_range_iff.1 $ bounded_of_compact $ compact_range f.continuous⟩
+
+@[simp] lemma mk_of_compact_apply [compact_space α] (f : C(α, β)) (a : α) :
+  mk_of_compact f a = f a :=
+rfl
 
 /-- If a function is bounded on a discrete space, it is automatically continuous,
 and therefore gives rise to an element of the type of bounded continuous functions -/
 def mk_of_discrete [discrete_topology α] (f : α → β) (hf : ∃C, ∀x y, dist (f x) (f y) ≤ C) :
   α →ᵇ β :=
```

Comment:
As far as I can tell, this lemma isn't used in the library.  But conjecturally, would it be more useful with the hypotheses `{C : ℝ} (h : ∀ x y:α, dist (f x) (f y) ≤ C)` rather than the existential?  Not sure if there's a rule of thumb for this.

Reply:
I like it, and even more with `(C : ℝ)`.